src/python/zquantum/core/utils_test.py

```diff
@@ -189,3 +191,18 @@ def test_noise_model_io(self):
 
         # Then
         self.assertEqual(noise_model, None)
+
+    def test_create_symbols_map(self):
+        # Given
+        symbol_1 = sympy.Symbol("alpha")
+        symbol_2 = sympy.Symbol("beta")
+        symbols = [symbol_1, symbol_2]
+        params = np.array([1, 2])
+        target_symbols_map = [(symbol_1, 1), (symbol_2, 2)]
+
+        # When
+        symbols_map = create_symbols_map(symbols, params)
+
+        # Then
+        self.assertEqual(symbols_map, target_symbols_map)
+
```

Comment:
Test cases that should exist:
- What happens when no symbols are present?
- What happens when no params are present?
- What happens when no symbols and no params are present?
- What happens when an unequal number of symbols and params are present?

Reply:
I think just testing whether the length of `symbols` and `params` matches should be enough.

Reply:
Adressed in [48a96b2](https://github.com/zapatacomputing/z-quantum-core/pull/42/commits/48a96b28f8d7021f8f7b8c07b82f89ee706be9e0)